Makefile

```
@@ -1,123 +1,110 @@
 
 ifeq ($(BOLOS_SDK),)
 $(error Environment variable BOLOS_SDK is not set)
 endif
+
 include $(BOLOS_SDK)/Makefile.defines
 
-#########
-#  App  #
-#########
-
-APPNAME    = Sia
-APPVERSION = 0.4.4
-ifeq ($(TARGET_NAME),TARGET_NANOS)
-ICONNAME=nanos_app_sia.gif
-else ifeq ($(TARGET_NAME),TARGET_NANOX)
-ICONNAME=nanox_app_sia.gif
-else ifeq ($(TARGET_NAME),TARGET_STAX)
-ICONNAME=stax_app_sia.gif
-endif
+########################################
+#        Mandatory configuration       #
+########################################
+# Application name
+APPNAME = "Sia"
 
-# The --path argument here restricts which BIP32 paths the app is allowed to derive.
-APP_LOAD_PARAMS = --path "44'/93'" --curve ed25519 $(COMMON_LOAD_PARAMS)
-ifeq ($(TARGET_NAME),TARGET_NANOS)
-APP_LOAD_PARAMS += --appFlags 0x40
-else ifeq ($(TARGET_NAME),TARGET_NANOX)
-APP_LOAD_PARAMS += --appFlags 0x40
-else ifeq ($(TARGET_NAME), TARGET_STAX)
-APP_LOAD_PARAMS += --appFlags 0x240
-endif
+# Application version
+APPVERSION_M = 1
+APPVERSION_N = 0
+APPVERSION_P = 0
+APPVERSION = "$(APPVERSION_M).$(APPVERSION_N).$(APPVERSION_P)"
 
+# Application source files
 APP_SOURCE_PATH += src
-SDK_SOURCE_PATH += lib_stusb lib_stusb_impl
-ifneq ($(TARGET_NAME),TARGET_STAX)
-SDK_SOURCE_PATH += lib_ux
-endif
-
-all: default
 
-load: all
-	python -m ledgerblue.loadApp $(APP_LOAD_PARAMS)
-
-delete:
-	python -m ledgerblue.deleteApp $(COMMON_DELETE_PARAMS)
-
-############
-# Platform #
-############
-
-DEFINES += OS_IO_SEPROXYHAL
-DEFINES += HAVE_SPRINTF
-DEFINES += HAVE_IO_USB HAVE_L4_USBLIB IO_USB_MAX_ENDPOINTS=7 IO_HID_EP_LENGTH=64 HAVE_USB_APDU
+# Application icons following guidelines:
+# https://developers.ledger.com/docs/embedded-app/design-requirements/#device-icon
+ICON_NANOS = icons/nanos_app_sia.gif
+ICON_NANOX = icons/nanox_app_sia.gif
+ICON_NANOSP = icons/nanos2_app_sia.gif
+ICON_STAX = icons/stax_app_sia.gif
+
+# Application allowed derivation curves.
+# Possibles curves are: secp256k1, secp256r1, ed25519 and bls12381g1
+# If your app needs it, you can specify multiple curves by using:
+# `CURVE_APP_LOAD_PARAMS = <curve1> <curve2>`
+CURVE_APP_LOAD_PARAMS = ed25519
+
+# Application allowed derivation paths.
+# You should request a specific path for your app.
+# This serve as an isolation mechanism.
+# Most application will have to request a path according to the BIP-0044
+# and SLIP-0044 standards.
+# If your app needs it, you can specify multiple path by using:
+# `PATH_APP_LOAD_PARAMS = "44'/1'" "45'/1'"`
+PATH_APP_LOAD_PARAMS = "44'/93'"
+
+# Setting to allow building variant applications
+# - <VARIANT_PARAM> is the name of the parameter which should be set
+#   to specify the variant that should be build.
+# - <VARIANT_VALUES> a list of variant that can be build using this app code.
+#   * It must at least contains one value.
+#   * Values can be the app ticker or anything else but should be unique.
+VARIANT_PARAM = COIN
+VARIANT_VALUES = BOL
```

```suggestion
VARIANT_VALUES = sia
```
To be iso with previous version.